cmd/timestamp-cli/app/verify.go

```diff
@@ -85,11 +93,139 @@ func runVerify() (interface{}, error) {
 		return nil, err
 	}
 
-	err = verification.VerifyTimestampResponse(tsrBytes, artifact, certPool)
+	opts, err := newVerifyOpts()
+	if err != nil {
+		return verifyCmdOutput{TimestampPath: tsrPath}, fmt.Errorf("failed to created VerifyOpts: %w", err)
+	}
+
+	err = verification.VerifyTimestampResponse(tsrBytes, artifact, certPool, opts)
 
 	return &verifyCmdOutput{TimestampPath: tsrPath}, err
 }
 
+func newVerifyOpts() (verification.VerifyOpts, error) {
+	opts := verification.VerifyOpts{}
+
+	oid, err := getOID()
+	if err != nil {
+		return verification.VerifyOpts{}, fmt.Errorf("failed to parse value from oid flag: %w", err)
+	}
+	if oid != nil {
+		opts.OID = oid
+	}
+
+	certPathFlagVal := viper.GetString("certificate")
+	if certPathFlagVal != "" {
+		cert, err := parseTSACertificate(certPathFlagVal)
+		if err != nil {
+			return verification.VerifyOpts{}, fmt.Errorf("failed to parse cert flag value from PEM file: %w", err)
+		}
+		opts.TSACertificate = cert
+	}
+
+	roots, intermediates, err := getRootAndIntermediateCerts()
+	if err != nil {
+		return verification.VerifyOpts{}, fmt.Errorf("failed to parse root and intermediate certs from cert-chain flag: %w", err)
+	}
+	if roots != nil && intermediates != nil {
+		opts.Roots = roots
+		opts.Intermediates = intermediates
+	}
+
+	nonce, err := getNonce()
+	if err != nil {
+		return verification.VerifyOpts{}, fmt.Errorf("failed to parse value from nonce flag: %w", err)
+	}
+	if nonce != nil {
+		opts.Nonce = nonce
+	}
+
+	commonNameFlagVal := viper.GetString("common-name")
+	if commonNameFlagVal != "" {
+		opts.CommonName = commonNameFlagVal
+	}
+
+	return opts, nil
+}
+
+func getNonce() (*big.Int, error) {
+	nonceFlagVal := viper.GetString("nonce")
+	if nonceFlagVal == "" {
+		return nil, nil
+	}
+
+	nonce := new(big.Int)
+	nonce, ok := nonce.SetString(nonceFlagVal, 10)
+	if !ok {
+		return nil, fmt.Errorf("failed to convert string to big.Int")
+	}
+	return nonce, nil
+}
+
+func getRootAndIntermediateCerts() ([]*x509.Certificate, []*x509.Certificate, error) {
+	certChainPEM := viper.GetString("cert-chain")
+	if certChainPEM == "" {
+		return nil, nil, nil
+	}
+
+	pemBytes, err := os.ReadFile(filepath.Clean(certChainPEM))
+	if err != nil {
+		return nil, nil, fmt.Errorf("error reading request from file: %w", err)
+	}
+
+	certs, err := cryptoutils.UnmarshalCertificatesFromPEM(pemBytes)
+	if err != nil {
+		return nil, nil, fmt.Errorf("failed to parse intermediate and root certs from PEM file: %w", err)
+	}
+
+	if len(certs) < 2 {
```

Comment:
We should change to len(certs) == 0 or len(certs) < 1, because you can provide only a root.